### DISCUSSION TITLE
Laravel app on Vapor

### DISCUSSION THREAD

**Discussion by dottely-code** [spatie/laravel-sitemap#481] [Q&A]
We have a Laravel app on Vapor and the storage is S3. How we can manage to write the sitemap.xml file directly to S3?

**Comment by nlehman-momentum** [comment]
I'm facing the same thing.  Did you ever figure this out @dottely-code ?

**Reply by dottely-code** [reply]
@nlehman-momentum  We are currently using the `writeToDisk` method of `SitemapIndex` class as there is no writeToDisk method on `SitemapGenerator` class and the `writeToFile` method that exists in this class uses `file_put_contents` under the hood that doesn't take into consideration the disk used by the framework of course.

**Comment by trevsewell** [comment]
You can save the sitemap contents directly to s3...

`Storage::put('sitemap.xml', $sitemap->render(), 'public');`

**Comment by joshcirre** [comment]
After storing it directly on S3, I just return the content directly in my routes:

`Route::get('/sitemap.xml', function () {
    $sitemap = Storage::disk('s3')->get('sitemap.xml');
    return response($sitemap, 200, [
        'Content-Type' => 'application/xml'
    ]);
});`
